src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java

```diff
@@ -189,13 +191,37 @@ public String apply(String originCreatedDate, int currentIndex, Translation tran
       }
       return format(FIELD_PATTERN, createdDateParam, publicationDate0Param, publicationDate1Param, languageParam);
     }
+  },
+
+  SET_INSTANCE_FORMAT_ID() {
+    private static final String REGEX = "--";
+
+    @Override
+    public String apply(String instanceFormatId, int currentIndex, Translation translation, ReferenceData referenceData, Metadata metadata) {
+      JsonObject entry = referenceData.getInstanceFormats().get(instanceFormatId);
+      if (entry == null) {
+        LOGGER.error("Instance format is not found by the given id: {}", instanceFormatId);
+        return StringUtils.EMPTY;
+      } else {
+        String instanceFormatIdValue = entry.getString("name");
+        List<String> instanceFormatsResult = Arrays.asList(instanceFormatIdValue.split(REGEX));
+        if (translation.getParameter(VALUE).equals("0") && isNotBlank(instanceFormatsResult.get(0))) {
+          return Arrays.asList(instanceFormatIdValue.split(REGEX)).get(0).trim();
```

Comment:
```suggestion
          return instanceFormatIdValue.split(REGEX))[0].trim();
```
this should work right? why do you need to convert it to an array?